examples/mpibench/Dockerfile.mpich

```diff
@@ -1,12 +1,13 @@
 # ch-test-scope: full
 FROM mpich
 
-RUN dnf install -y which
+RUN    dnf install -y which \
```

Comment:
`dnf` is missing `--setopt=install_weak_deps=false` here?

Reply:
Since it's only installing `which` I didn't believe it had additional weak dependencies it was bringing in anyway, I may have missed them though.

Reply:
I think it would be good to be consistent about adding the no weak dependencies options for both yum and apt, since it will keep the images smaller and avoid wondering why it's missing in some cases.

Reply:
After thinking about it more I agree, it keeps folks from feeling like they need to lookup if a package has weak dependencies or not and consistency is good.